### ISSUE TITLE
Is the CFG ratio flipped?

### ISSUE THREAD

**Issue by Flux9665** [haidog-yaqub/MeanFlow#10]
Hi, thank you for providing this implementation!

I am not sure if I am misundertanding something here, but it seems to me that the CFG ratio is flipped. 

https://github.com/haidog-yaqub/MeanFlow/blob/ab57f94af54dcd50a0ab80acebde422a74625306/meanflow.py#L145

creates a mask that is True in `self.cfg_ratio` cases and False otherwise.

https://github.com/haidog-yaqub/MeanFlow/blob/ab57f94af54dcd50a0ab80acebde422a74625306/meanflow.py#L156

This mask is applied here and will take the value from `v_hat` wherever the mask is False (90% of cases using default settings) and `v` wherever the mask is True (10% of the cases). Should it not be the other way around?

**Comment by haidog-yaqub** [comment]
> Hi, thank you for providing this implementation!
> 
> I am not sure if I am misundertanding something here, but it seems to me that the CFG ratio is flipped.
> 
> [MeanFlow/meanflow.py](https://github.com/haidog-yaqub/MeanFlow/blob/ab57f94af54dcd50a0ab80acebde422a74625306/meanflow.py#L145)
> 
> Line 145 in [ab57f94](/haidog-yaqub/MeanFlow/commit/ab57f94af54dcd50a0ab80acebde422a74625306)
> 
>  cfg_mask = torch.rand_like(c.float()) < self.cfg_ratio 
> creates a mask that is True in `self.cfg_ratio` cases and False otherwise.
> 
> [MeanFlow/meanflow.py](https://github.com/haidog-yaqub/MeanFlow/blob/ab57f94af54dcd50a0ab80acebde422a74625306/meanflow.py#L156)
> 
> Line 156 in [ab57f94](/haidog-yaqub/MeanFlow/commit/ab57f94af54dcd50a0ab80acebde422a74625306)
> 
>  v_hat = torch.where(cfg_mask, v, v_hat) 
> This mask is applied here and will take the value from `v_hat` wherever the mask is False (90% of cases using default settings) and `v` wherever the mask is True (10% of the cases). Should it not be the other way around?

Hi there, thank you for the good question. This only applies to the unconditional elements where you want to use the original v rather than v_hat = self.w * v + (1 - self.w) * u_t. So It is only applied to the masked elements.
